### PR TITLE
Naynjain/edk2fixes v3

### PR DESCRIPTION
**Pull request by naynajain**
This has changes related to:
clearing timestamp for physical presence and hw key hash.
renamed the function clear_all_os_keys() to reset_keystore()
renamed the file edk2-compat-clear-keys.c to edk2-compat-reset.c
Updated test file with stub functions
Added validate_esl_list() check to be called when processing authentication of new ESL.
Removed is_single_pk() call as we now have validate_esl_list()

### CODE REVIEW

File: libstb/secvar/backend/edk2-compat.c [hunk 1]
```diff
@@ -163,10 +165,16 @@ static int edk2_compat_process(void)
 
 static int edk2_compat_post_process(void)
 {
+	struct secvar_node *hwvar;
 	printf("setup mode is %d\n", setup_mode);
 	if (!setup_mode) {
 		secvar_set_secure_mode();
 		prlog(PR_INFO, "Enforcing OS secure mode\n");
+		hwvar = find_secvar("HWKH", 5, &variable_bank);
```
Comment (erichte-ibm) [inline libstb/secvar/backend/edk2-compat.c:182]: Probably should have a short comment explaining why we are doing this: "hw-key-hash is no longer needed after this point, and doesn't need to be exposed"

File: libstb/secvar/backend/edk2-compat.c [hunk 2]
```diff
@@ -102,16 +95,25 @@ static int edk2_compat_process(void)
         /* Check if physical presence is asserted */
         if (is_physical_presence_asserted()) {
                 prlog(PR_INFO, "Physical presence asserted to clear OS Secure boot keys\n");
-                clear_all_os_keys();
+                rc = reset_keystore();
+		if (rc) {
```
Comment (erichte-ibm) [inline libstb/secvar/backend/edk2-compat.c]: Inconsistent whitespace, no big deal, will probably do a mass find-and-replace when building the patches

File: libstb/secvar/backend/edk2-compat-process.c [hunk 3]
```diff
@@ -596,6 +627,11 @@ int process_update(struct secvar_node *update, char **newesl,
 	}
 	memcpy(*newesl, update->var->data + auth_buffer_size, *new_data_size);
 
+	/* Validate the new ESL is in right format */
+	rc = validate_esl_list(update->var->key, *newesl, *new_data_size);
```
Comment (erichte-ibm) [inline libstb/secvar/backend/edk2-compat-process.c:651]: Candidate for calling the full `.validate()` function here, and including `validate_esl_list()` in there

**Comment by erichte-ibm** [comment]
Overall suggestion: functions that aren't used outside of their respective `.c` files can be marked as static and doesn't need to be exposed in the header (which is the majority of the functions in `edk2-process.h`).

File: libstb/secvar/backend/edk2-compat-process.c (hunk 3 above)
Comment (naynajain) [reply libstb/secvar/backend/edk2-compat-process.c:651]: I thought edk2-compat-process.c is sort of internal file which hooks will call. And hooks to be called by secvar_main or platform thing.. I will not prefer edk2-compat-process calling hooks internally. I think it is .validate() hook that should call validate_esl_list ().

Also, .validate() hook will be doing much more.. like we can add a function validate_auth().. that does auth command validation.. So for certain I don't want to call .validate() hook inside edk2-compat-process.c process_update() function

Comment (erichte-ibm) [reply libstb/secvar/backend/edk2-compat-process.c:651]: Why would we not want to reuse code? If its doing much more, maybe we should call the validate function before we attempt to process the update at all, to ensure it works.

The hook doesn't need to be called from the driver struct, but I do think we should at least reuse either the function exposed in the struct, or the functions it calls. I'll take a look at this when I do a full file review

Comment (naynajain) [reply libstb/secvar/backend/edk2-compat-process.c:651]: Exactly we need to reuse the code.. and for that reason I think. .validate() hook should call validate_esl_list().  I don't want to mix up the interfaces. Well,  I am not sure if that explains the reason so well..but I still do not feel calling .validate() hook in edk2-compat-process.c  So.. sure go ahead and review it.. and if you still want to do it... its ok.. just that it is my choice not to do so..

**Pull request merged**